pkg/kubelet/cm/devicemanager/plugin/v1beta1/stub.go

```diff
@@ -76,20 +81,36 @@ func defaultAllocFunc(r *pluginapi.AllocateRequest, devs map[string]pluginapi.De
 	return &response, nil
 }
 
+// stubRegisterControlFunc is the function called when a registration request is received from Kubelet
+type stubRegisterControlFunc func() bool
+
+func defaultRegisterControlFunc() bool {
+	return true
+}
+
 // NewDevicePluginStub returns an initialized DevicePlugin Stub.
 func NewDevicePluginStub(devs []*pluginapi.Device, socket string, name string, preStartContainerFlag bool, getPreferredAllocationFlag bool) *Stub {
+
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		klog.ErrorS(err, "Watcher creation failed")
+		panic(err)
```

Comment:
how do we decide between `panic` and returning error? Is the kubelet watcher a critical functionality?

Reply:
The watcher here watches for changes to a directory where all the device plugin sockets are placed which is important for us to monitor to identify kubelet restarts. When a new instance of kubelet starts, it deletes all Unix sockets under the directory being watched here (/var/lib/kubelet/device-plugins). In case the watcher fails it seems critical enough to me to surface this error. I chose panic here because the alternative here (like you suggested) is to return an error but that would impact device plugin implementations that are using this `NewDevicePluginStub` to bootstrap their respective plugins. I am aware of at least [one device plugin implementation](https://github.com/k8stopologyawareschedwg/sample-device-plugin/blob/main/pkg/deviceplugin/deviceplugin.go) that relies on this and didn't want these plugins to break in case they upgrade to Kubernetes latest toolchain.

If you feel strongly about it, please let me know and I will make updates to return an error here instead and let the plugins handle the error themselves.

Reply:
sorry, got interrupted on this PR review and then lost track of it. Coming back here.

So you are saying that without the kubelet restart monitoring functionality this stub doesn't make sense to use? If so, `panic` is OK

Reply:
Yep, pretty much. A device  plugin is expected to handle kubelet restarts as has been documented [here](https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/device-plugins/#handling-kubelet-restarts).